src/vivarium/framework/results/manager.py

```diff
@@ -180,22 +180,21 @@ def gather_results(self, event_name: str, event: Event) -> None:
         with 0.0.
         """
         population = self._prepare_population(event)
-        for results_group, measure in self._results_context.gather_results(
+        for results_group, measure, updater in self._results_context.gather_results(
             population, event_name
         ):
             if results_group is not None:
                 if measure is None:
                     raise ValueError(
                         f"There is a results group {results_group} but no corresponding measure"
                     )
-                # Look for extra columns in the results_group and initialize with 0.
-                extra_cols = list(
-                    set(results_group.columns) - set(self._raw_results[measure].columns)
+                if updater is None:
```

Comment:
When can `measure` or `updater` be `None`? This should error on registration rather than during `gather_results`.

Reply:
I don't think they can be none. Will remove these in next PR

Reply:
Oh actually @rmudambi I remember now - I had to add these to proprly handle the type hinting since `self._results_context.gather_results` can indeed yield None, None, None (when there is no population after filtering). Kind of a sad fact of type hints I think.

Reply:
Oh, duh, I can check that all three are not none. I'll fix in next PR